src/test/java/org/folio/dew/batch/bursarfeesfines/service/BursarFilterEvaluatorTest.java

```diff
@@ -0,0 +1,305 @@
+
+    assertTrue(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterLocation)
+    );
+  }
+
+  @Test
+  void filterAccountByPatronGroup() {
+    UUID patronGroupID = UUID.randomUUID();
+    BursarExportFilterPatronGroup filterPatronGroup = new BursarExportFilterPatronGroup();
+    filterPatronGroup.setPatronGroupId(patronGroupID);
+
+    User user = new User();
+    user.setPatronGroup(patronGroupID.toString());
+
+    AccountWithAncillaryData accountWithAncillaryData = AccountWithAncillaryData
+      .builder()
+      .account(null)
+      .user(user)
+      .item(null)
+      .build();
+
+    assertTrue(
+      BursarFilterEvaluator.evaluate(
+        accountWithAncillaryData,
+        filterPatronGroup
+      )
+    );
+  }
+
+  @Test
+  void filterAccountByServicePoint() {
+    UUID servicePointID = UUID.randomUUID();
+    BursarExportFilterServicePoint filterServicePoint = new BursarExportFilterServicePoint();
+    filterServicePoint.setServicePointId(servicePointID);
+
+    Item item = new Item();
+    item.setInTransitDestinationServicePointId(servicePointID.toString());
+
+    AccountWithAncillaryData accountWithAncillaryData = AccountWithAncillaryData
+      .builder()
+      .account(null)
+      .user(null)
+      .item(item)
+      .build();
+
+    assertTrue(
+      BursarFilterEvaluator.evaluate(
+        accountWithAncillaryData,
+        filterServicePoint
+      )
+    );
+  }
+
+  @Test
+  void evaluateNegationFilter() {
+    BursarExportFilterNegation filterNegation = new BursarExportFilterNegation();
+    filterNegation.setCriteria(new BursarExportFilterPass());
+
+    AccountWithAncillaryData accountWithAncillaryData = AccountWithAncillaryData
+      .builder()
+      .account(null)
+      .user(null)
+      .item(null)
+      .build();
+
+    assertFalse(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterNegation)
+    );
+  }
+
+  @Test
+  void evaluateConditionalFilters() {
+    BursarExportFilterCondition filterCondition = new BursarExportFilterCondition();
+
+    BursarExportFilterPass filterPass = new BursarExportFilterPass();
+
+    BursarExportFilterNegation notFilterPass = new BursarExportFilterNegation();
+    notFilterPass.setCriteria(filterPass);
+
+    AccountWithAncillaryData accountWithAncillaryData = AccountWithAncillaryData
+      .builder()
+      .account(null)
+      .user(null)
+      .item(null)
+      .build();
+
+    filterCondition.addCriteriaItem(filterPass);
+    filterCondition.addCriteriaItem(notFilterPass);
+
+    // testing AND filter
+    filterCondition.setOperation(BursarExportFilterCondition.OperationEnum.AND);
+    assertFalse(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterCondition)
+    );
+
+    // testing OR filter
+    filterCondition.setOperation(BursarExportFilterCondition.OperationEnum.OR);
+    assertTrue(
+      BursarFilterEvaluator.evaluate(accountWithAncillaryData, filterCondition)
+    );
+  }
+}
```

Comment:
There are some coverage gaps (4 lines, 8 branches), below is a report from jacoco.  To test this out yourself, you can run:

```
mvn test jacoco:report
```

This will run all tests (requires Docker to be running, too) and creates the output `target/site/jacoco/index.html`.  If you only want to run one test (these unit ones don't require Docker):

```
mvn test jacoco:report -Dtest=BursarFilterEvaluatorTest
```

<img width="756" alt="image" src="https://github.com/ualibweb/mod-data-export-worker/assets/8005215/59cd0016-15b2-4181-b0bd-afac9b53ddf4">
<img width="644" alt="image" src="https://github.com/ualibweb/mod-data-export-worker/assets/8005215/60e0b095-e260-4539-92ef-d2f1be562bfb">

You may have to get creative with the last else; one way you could test it is to declare a class (in the test package) that extends `BursarExportFilter` and pass that in

Reply:
@ncovercash So I'm thinking of creating a `helpers` package in test/java/org.folio.dew that will contain helper classes like the one we would need to handle the last `else`. Is that an ideal location to have the package in?